backend/hitas/views/apartment.py

```diff
@@ -366,38 +427,73 @@ class ApartmentViewSet(HitasModelViewSet):
-            "purchase_price",
-            "first_purchase_date",
-            "second_purchase_date",
-            "additional_work_during_construction",
-            "loans_during_construction",
-            "interest_during_construction",
-            "debt_free_purchase_price_during_construction",
-            "notes",
-            "completion_date",
-            "building__uuid",
-            "building__real_estate__uuid",
-            "apartment_type__uuid",
-            "apartment_type__value",
-            "apartment_type__legacy_code_number",
-            "apartment_type__description",
-            "building__real_estate__housing_company__uuid",
-            "building__real_estate__housing_company__display_name",
-            "building__real_estate__housing_company__postal_code__value",
-            "building__real_estate__housing_company__postal_code__city",
+        return (
+            self.get_list_queryset()
+            .only(
+                "uuid",
+                "state",
+                "surface_area",
+                "street_address",
+                "apartment_number",
+                "floor",
+                "stair",
+                "share_number_start",
+                "share_number_end",
+                "debt_free_purchase_price",
+                "primary_loan_amount",
+                "purchase_price",
+                "first_purchase_date",
+                "second_purchase_date",
+                "additional_work_during_construction",
+                "loans_during_construction",
+                "interest_during_construction",
+                "debt_free_purchase_price_during_construction",
+                "notes",
+                "completion_date",
+                "building__uuid",
+                "building__real_estate__uuid",
+                "apartment_type__uuid",
+                "apartment_type__value",
+                "apartment_type__legacy_code_number",
+                "apartment_type__description",
+                "building__real_estate__housing_company__uuid",
+                "building__real_estate__housing_company__display_name",
+                "building__real_estate__housing_company__postal_code__value",
+                "building__real_estate__housing_company__postal_code__city",
+            )
+            .extra(
+                select={
+                    "cpi": self.select_index("constructionpriceindex", pre2005=False),
+                    "cpi_pre2005": self.select_index("constructionpriceindexpre2005", pre2005=True),
+                    "mpi": self.select_index("marketpriceindex", pre2005=False),
+                    "mpi_pre2005": self.select_index("marketpriceindexpre2005", pre2005=True),
+                    "sapc": """
+    SELECT ROUND(a.surface_area * sapc.value)
+    FROM hitas_apartment AS a
+    LEFT JOIN hitas_surfaceareapriceceiling AS sapc
+        ON sapc.month = DATE_TRUNC('month', NOW())
+    WHERE a.id = hitas_apartment.id
+""",
```

Comment:
Is it not possible to do this with Django ORM, or is there some benefit we gain from doing this with raw SQL?

Reply:
see above